samples/sample_script.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env sh
+
+# set -xv
+
+# show usage and quit
+function usage()
+{
+  printf "usage: %s <command>\n" $(basename $0) >&2
+  printf "\n" >&2
+  printf "    commands are:\n"
+  printf "        download [<version>] - downloads and mounts all example data\n" >&2
+  printf "                               gets latest version unless <version> supplied\n" >&2
+  printf "        remove - removes all example data\n" >&2
+  printf "        version - show local version\n" >&2
+  printf "        versions - list available versions\n" >&2
+  exit 2
+}
+
+
+# complain and quit
+function fail_out()
+{
+  printf "Failed! %s\n" "$@" >&2
+  exit 2
+}
+
+
+# check that we have the expected enlistment directory; complain and fail if not
+function ensure_enlistment()
+{
+  if [ ! -d $target_path/.git ]; then
+     printf "no examples collection at %s; dowloading ..." $target_path >&2
+     do_download
+  fi
+}
+
+
+# clone the git repo, don't report progress but DO report errors
+function do_download()
+{
+  if [ -d $target_path/.git ]; then
+    printf "examples collection already exists at %s\n" $target_path >&2
+  else
+    git clone --quiet $git_root_url $target_path || fail_out "Couldn't clone examples repository"
+    printf "examples downloaded to $target_path\n"
+  fi
+  [ ! -z "$1" ] && do_checkout_version "$1"
+}
```

Comment:
Docs above say that it will get the latest version if no arg is given.  I think this code does not do that.  (1) if the collection is already checked out, it doesn't fetch the latest.  (2) if there is no argument, it doesn't fetch the latest.

Reply:
Fixed! We'll get the current version if not specified in all cases; and use the supplied version in all cases if it is supplied.